src/main/java/pl/javastart/task/TimeCalculator.java

```diff
@@ -0,0 +1,123 @@
+            if (Character.isDigit(timeUnit.charAt(0))) {
+                secondMultiply = Integer.parseInt(split[i]);
+                i++;
+            }
+            timeUnit = split[i];
+            timeUnits.add(new DataUnitFromUser(sign, multiply, secondMultiply, timeUnit));
+        }
+        System.out.println(timeUnits);
+        return timeUnits;
+    }
+
+    public static LocalDateTime createTimeResult(String dataInput) {
+        List<DataUnitFromUser> dateList = createDateList(dataInput);
+        LocalDateTime localDateTime = null;
+        for (DataUnitFromUser unit : dateList) {
+            if (Objects.equals(unit.getTimeUnit(), YEAR)) {
+                if (Objects.equals(unit.getSign(), PLUS)) {
+                    localDateTime = LocalDateTime.now().plusYears(unit.getMultiplyNumber());
+                } else {
+                    localDateTime = LocalDateTime.now().minusYears(unit.getMultiplyNumber());
+                }
+            } else if (Objects.equals(unit.getTimeUnit(), MONTH)) {
+                if (Objects.equals(unit.getSign(), PLUS)) {
+                    localDateTime = LocalDateTime.now().plusMonths(unit.getMultiplyNumber());
+                } else {
+                    localDateTime = LocalDateTime.now().minusMonths(unit.getMultiplyNumber());
+                }
+            } else if (Objects.equals(unit.getTimeUnit(), DAY)) {
+                if (Objects.equals(unit.getSign(), PLUS)) {
+
+                    localDateTime = LocalDateTime.now().plusDays(unit.getMultiplyNumber());
+                } else {
+                    localDateTime = LocalDateTime.now().minusDays(unit.getMultiplyNumber());
+                }
+            } else if (Objects.equals(unit.getTimeUnit(), HOUR)) {
+                if (Objects.equals(unit.getSign(), PLUS)) {
+                    localDateTime = LocalDateTime.now().plusHours(unit.getMultiplyNumber());
+                } else {
+                    localDateTime = LocalDateTime.now().minusHours(unit.getMultiplyNumber());
+                }
+            } else if (Objects.equals(unit.getTimeUnit(), MINUTE)) {
+                if (Objects.equals(unit.getSign(), PLUS)) {
+                    localDateTime = LocalDateTime.now().plusMinutes(unit.getMultiplyNumber());
+                } else {
+                    localDateTime = LocalDateTime.now().minusHours(unit.getMultiplyNumber());
+                }
+            } else if (Objects.equals(unit.getTimeUnit(), SECOND)) {
+                if (Objects.equals(unit.getSign(), PLUS)) {
+                    localDateTime = LocalDateTime.now().plusSeconds(unit.getMultiplyNumber());
+                } else {
+                    localDateTime = LocalDateTime.now().minusSeconds(unit.getMultiplyNumber());
+                }
+            }
+        }
+        return localDateTime;
+    }
+
+    static class DataUnitFromUser {
+        String sign;
+        int multiplyNumber;
+        int secondMultiply;
```

Comment:
To jest główny problem Twojego rozwiązania, że osobno przechowujesz 1 i 2 cyfrę liczby. W zadaniu nie ma jednak ograniczenia, które mówiłoby, że nie może być też np. +123m, albo +1234m
a oczywiście nie będziemy tworzyli klasy, z 30 polami "na zapas"